### PR TITLE
update documentation with examples comment specifications markdown pages

### LINKED ISSUE #742: [DOCUMENTATION] update PPL documentation and readme to reflect updated status
**Is your feature request related to a problem?**
Update PPL documentations and readme to reflect updated status:
 - roadmap
 - existing ppl commands
 - document new / planned commands

**Do you have any additional context?**

### PR DESCRIPTION
**Pull request by YANG-DB**
### Description
update documentation with examples comment specifications markdown pages
 - PPL main readme doc
 - PPL lang examples doc
 - dedicated command doc

### Issues Resolved
#742 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/ppl-lang/ppl-dedup-command.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+- `source = table | dedup 1 a,b keepempty=true | fields a,b,c`
+- `source = table | dedup 2 a | fields a,b,c`
+- `source = table | dedup 2 a,b | fields a,b,c`
+- `source = table | dedup 2 a keepempty=true | fields a,b,c`
+- `source = table | dedup 2 a,b keepempty=true | fields a,b,c`
+- `source = table | dedup 1 a consecutive=true| fields a,b,c` (Consecutive deduplication is unsupported)
+
+### Limitation:
+
+**Spark Support** (3.4)
```
Comment (anasalkouz) [inline docs/ppl-lang/ppl-dedup-command.md:129]: I thought dedup 2 requires spark 3.5

File: docs/ppl-lang/ppl-rename-command.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    | an   | emp     |
+    |------+---------|
+    | 1    | Pyrami  |
+    | 6    | Netagy  |
+    | 13   | Quility |
+    | 18   | null    |
+    +------+---------+
+
+### Limitation:
+- `rename` command needs spark version >= 3.4
```
Comment (anasalkouz) [inline docs/ppl-lang/ppl-rename-command.md:50]: 3.5?

Comment (YANG-DB) [reply docs/ppl-lang/ppl-rename-command.md:50]: from version 3.4 and above

Comment (LantaoJin) [reply docs/ppl-lang/ppl-rename-command.md:50]: @YANG-DB PPL commands which require plan `DataFrameDropColumns` need the version from 3.5.1

Comment (LantaoJin) [reply docs/ppl-lang/ppl-rename-command.md:50]: https://issues.apache.org/jira/browse/SPARK-45770

File: docs/ppl-lang/ppl-dedup-command.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+:     +- Window [row_number() windowspecdefinition('a, 'b, 'a ASC NULLS FIRST, 'b ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _row_number_], ['a, 'b], ['a ASC NULLS FIRST, 'b ASC NULLS FIRST]
+:        +- Filter (isnotnull('a) AND isnotnull('b))
+:           +- Project
+:              +- UnresolvedRelation
++- Filter (isnull('a) OR isnull('b))
+   +- Project
+      +- UnresolvedRelation
+```
+
+ - this `dedup` command with `allowedDuplication > 1` feature needs spark version >= 3.4
```
Comment (LantaoJin) [inline docs/ppl-lang/ppl-dedup-command.md:157]: 3.5.1

**Pull request merged**